### DISCUSSION TITLE
exclude files but keep folder structure

### DISCUSSION THREAD

**Discussion by Nethan** [borgbackup/borg#5940] [Ideas]
Hi,

A option to exclude files but let folder structure intact.

I want exclude the Logfiles /var/log (But after a restore i need the folders in /var/log like /var/log/mysql ..)
My workaround 
    --pattern=-/var/log/\*.gz \
    --pattern=-/var/log/\*.log
works (as long no folder has a name like something.log) but it would be nice if there is a switch like --exclude-files-only /var/log/

**Comment by justin2004** [comment]
can you just put an empty file in the folders you want to preserve?

e.g.  `touch /var/log/mysql/keepme`

**Comment by ThomasWaldmann** [comment]
https://borgbackup.readthedocs.io/en/stable/usage/create.html

See "exclusion options" there, esp.  --exclude-if-present and --keep-tag-files.

borg create --exclude-if-present .nobackup --keep-tag-files ...

So this would exclude any directory that has a .nobackup file inside, but it would backup the .nobackup file itself, so it will be present again after a restore (and also the containing directory would be present, of course).
